### PR TITLE
Alertmanager configured route metric

### PR DESCRIPTION
**Pull request by paulfantom**
Idea is to have a metric which can be gathered by telemetry and tell us if clients use alertmanager and send notifications/alerts at all.

This will return simple binary value.

/cc @smarterclayton @simonpasquier @s-urbaniak

### CODE REVIEW

**Comment by smarterclayton** [comment]
This looks sufficient to test the outcome for me.

Can we add a warning alert with a 1h interval that suggests the user configure notifications?  Not having notification sinks configured to me *seems* like it crosses the warning threshold in that without notification sinks alert manager is at best a dashboard function.  If there's controversy on that, we can take it as a separate PR, otherwise I'd suggest doing it here.

@brancz @s-urbaniak @LiliC opinions?

**Comment by paulfantom** [comment]
> Can we add a warning alert with a 1h interval that suggests the user configure notifications

Where would that be sent when there are no alertmanager routes?

**Comment by s-urbaniak** [comment]
@smarterclayton I guess this is a good suggestion. While we have a watchdog alert by default, it still doesn't really kick in until you have fully configured alertmanager. A concrete alert showing up in console highlighting the fact of an unconfigured monitoring stack is probably a good addition, wdyt @LiliC @brancz.

**Comment by smarterclayton** [comment]
Plus we can use the presence of that alert firing to attach the UX for
"hey, should configure a notification sink" vs using some other store for
that value.  If you don't want it, silence it.  Either way you learn how to
use alerts.

**Comment by s-urbaniak** [comment]
@paulfantom ping, this needs a rebase

**Comment by paulfantom** [comment]
rebased

File: assets/prometheus-k8s/rules.yaml [hunk 1]
```diff
@@ -356,6 +356,8 @@ spec:
         label_node_hyperthread_enabled, label_node_openshift_io_os_id,label_node_role_kubernetes_io_master,label_node_role_kubernetes_io_infra)
         ) by (label_kubernetes_io_arch, label_node_hyperthread_enabled, label_node_openshift_io_os_id,label_node_role_kubernetes_io_master,label_node_role_kubernetes_io_infra)
       record: node_role_os_version_machine:cpu_capacity_sockets:sum
+    - expr: clamp_max(sum(alertmanager_notifications_total),1)
+      record: alertmanager_routing_enabled
```
Comment (brancz) [inline assets/prometheus-k8s/rules.yaml]: This doesn’t follow naming best practices.

Comment (paulfantom) [reply assets/prometheus-k8s/rules.yaml]: Do you think something like `cluster:alertmanager_routing_enabled:max` would be better?

Comment (brancz) [reply assets/prometheus-k8s/rules.yaml]: yes

**Comment by smarterclayton** [comment]
When the alert is added can you link it to here?

/cherry-pick release-4.3

**Pull request merged**